File: lagom/examples/bc1-to-bc2/bc1-to-bc2-impl/src/main/scala/com/apuex/sales/mapping/bc1ToBc2/OrderService.scala
package com.apuex.sales.mapping.bc1ToBc2

import javax.inject._

import akka.{Done, NotUsed}
import akka.stream.scaladsl.Source
import com.lightbend.lagom.scaladsl.api.{Descriptor, Service, ServiceCall}
       
trait OrderService extends Service {
  /**
    * Subscribe from event stream with offset.
    *
    * @see [[akka.persistence.query.Offset]]
    * @see [[akka.persistence.query.TimeBasedUUID]]
    * @param offset timed-uuid specifies start position
    * @return
    */
  def events(offset: Option[String]): ServiceCall[Source[String, NotUsed], Source[String, NotUsed]]
         
  def retrieve(): ServiceCall[RetrieveOrderCmd, OrderVo]

  
  override final def descriptor: Descriptor = {
    import Service._
  
    named("order")
      .withCalls(
        pathCall("/api/events", events _)
      ).withAutoAcl(true)
  }
       
}

File: lagom/examples/bc1-to-bc2/bc1-to-bc2-impl/src/main/scala/com/apuex/sales/mapping/bc1ToBc2/OrderService.java
package com.apuex.sales.mapping.bc1ToBc2;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;

import java.util.Optional;

import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.pathCall;

public interface OrderService extends Service {

  ServiceCall<RetrieveOrderCmd, OrderVo> retrieve();

  /**
   * Subscribe from event stream with offset.
   *
   * @param offset timed-uuid or sequence specifies start position
   * @return
   */
  ServiceCall<akka.stream.javadsl.Source<String, NotUsed>, akka.stream.javadsl.Source<String, NotUsed>> events(Optional<String> offset);

  @Override
  default Descriptor descriptor() {
    return named("order")
        .withCalls(
            pathCall("/api/events?offset", this::events),
            pathCall("/api/retrieve-order", this::retrieve)
        ).withAutoAcl(true);
  }
}
